src/main/java/com/uid2/shared/auth/AuthorizableStore.java

```diff
@@ -0,0 +1,75 @@
+package com.uid2.shared.auth;
+
+import com.uid2.shared.secret.KeyHasher;
+
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+
+public class AuthorizableStore<T extends IAuthorizable> {
+    private class AuthorizableStoreSnapshot {
+        private final Map<ByteBuffer, T> authorizables;
+        private final Collection<byte[]> salts;
+
+        public AuthorizableStoreSnapshot(Collection<T> authorizables) {
+            this.authorizables = authorizables.stream()
+                    .filter(a -> a.getKeyHash() != null) // TODO: remove this filter when all keys have hashes
+                    .collect(Collectors.toMap(
+                            a -> ByteBuffer.wrap(Base64.getDecoder().decode(a.getKeyHash())),
+                            a -> a
+                    ));
+            this.salts = authorizables.stream()
+                    .filter(a -> a.getKeySalt() != null) // TODO: remove this filter when all keys have salts
```

Comment:
Admins, clients and operators should now all have key salt, so this should be good to remove

Reply:
Removed